shared/src/main/scala/mlscript/TypeSimplifier.scala

```diff
@@ -282,4 +282,107 @@ trait TypeSimplifier { self: Typer =>
     
   }
   
+  def reconstructClassTypes(st: SimpleType, pol: Bool, ctx: Ctx): SimpleType = {
+    
+    val renewed = MutMap.empty[TypeVariable, TypeVariable]
+    
+    def renew(tv: TypeVariable): TypeVariable =
+      if (tv.lowerBounds.isEmpty && tv.upperBounds.isEmpty) tv
+      else renewed.get(tv) match {
+        case S(tv2) => tv2
+        case N =>
+          val tv2 = freshVar(tv.prov, tv.nameHint)(0)
```

Comment:
Yes you're right. Currently the type simplifier uses `0` everywhere because as you say it's just for printing, but it's true we should be using `tv.level`.